### PR TITLE
Hubspot response filter & better documentation

### PR DESCRIPTION
**Pull request by AlexTeusz**
I updated the README for a better understanding of the Custom Module. Furthermore the line 	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only'; was added to get the full API response in the Cognigy Context. 

To test this model, create a project in the Cognigy.AI and upload this Custom Module to the Interaction Framework.

### CODE REVIEW

File: modules/hubspot/README.md [hunk 1]
```diff
@@ -1,8 +1,196 @@
...
+      "property": "firstname",
+      "value": "CM TEST"
+    }
+  ]
+}
+```
+
+If the creation was successful, the API  returns the new Contact in a JSON format. 
+
+# Node: searchContact
```
Comment (robinschuster) [inline modules/hubspot/README.md]: we are searching through contacts, or finding a contact. rename to "search contacts" or "find contact"

File: modules/hubspot/src/module.ts [hunk 2]
```diff
@@ -14,20 +14,26 @@ async function findContactByEmail(input: IFlowInput, args: { secret: CognigySecr
 	if (!args.email) return Promise.reject("No email defined.");
 
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
 	let result = {};
 
 	return new Promise((resolve, reject) => {
 		hubspot.contacts.getByEmail(args.email, (err, res) => {
+
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: unnecessary line break

File: modules/hubspot/src/module.ts [hunk 3]
```diff
@@ -14,20 +14,26 @@ async function findContactByEmail(input: IFlowInput, args: { secret: CognigySecr
...
 			if (err) {
 				// if an error was triggered, either reject or write error to store
 				if (args.stopOnError) { reject(err); return; }
 				else result = { "error": err.message };
 			} else {
 				const properties = (args.properties) ? args.properties.split(",") : [];
 				if (properties.indexOf("vid") === -1) properties.push("vid");
-				Object.keys(res).forEach((key) => {
+
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: unused line break

File: modules/hubspot/src/module.ts [hunk 4]
```diff
@@ -14,20 +14,26 @@ async function findContactByEmail(input: IFlowInput, args: { secret: CognigySecr
 	if (!args.email) return Promise.reject("No email defined.");
 
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: you can access this directly with `hubspot.qs.propertyMode`

File: modules/hubspot/src/module.ts [hunk 5]
```diff
@@ -14,20 +14,26 @@ async function findContactByEmail(input: IFlowInput, args: { secret: CognigySecr
...
 			if (err) {
 				// if an error was triggered, either reject or write error to store
 				if (args.stopOnError) { reject(err); return; }
 				else result = { "error": err.message };
 			} else {
 				const properties = (args.properties) ? args.properties.split(",") : [];
 				if (properties.indexOf("vid") === -1) properties.push("vid");
-				Object.keys(res).forEach((key) => {
+
+				Object.keys(res["properties"]).forEach((key) => {
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `res.properties`

File: modules/hubspot/src/module.ts [hunk 6]
```diff
@@ -14,20 +14,26 @@ async function findContactByEmail(input: IFlowInput, args: { secret: CognigySecr
...
+
 					// if key isn't in the defined properties, delete it
-					if (properties.indexOf(key) === -1) delete res[key];
+					if (properties.indexOf(key) === -1) {
+						delete res["properties"][key]
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `res.properties[key]`

File: modules/hubspot/src/module.ts [hunk 7]
```diff
@@ -68,6 +75,7 @@ async function updateContact(input: IFlowInput, args: { secret: CognigySecret, v
 	if (!args.data) return Promise.reject("No data defined.");
 
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `hubspot.qs.propertyMode`

File: modules/hubspot/src/module.ts [hunk 8]
```diff
@@ -105,7 +113,9 @@ async function updateContact(input: IFlowInput, args: { secret: CognigySecret, v
 async function createContact(input: IFlowInput, args: { secret: CognigySecret, data: any, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<any> {
 	if (!args.secret || !args.secret.apiKey) return Promise.reject("Secret not defined or invalid.");
 	if (!args.data) return Promise.reject("No data defined.");
+
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: hubspot.qs.propertyMode

File: modules/hubspot/src/module.ts [hunk 9]
```diff
@@ -144,7 +154,9 @@ async function createContact(input: IFlowInput, args: { secret: CognigySecret, d
 async function searchContact(input: IFlowInput, args: { secret: CognigySecret, query: string, properties: string, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<any> {
 	if (!args.secret || !args.secret.apiKey) return Promise.reject("Secret not defined or invalid.");
 	if (!args.query) return Promise.reject("No query defined.");
+
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: hubspot.qs.propertyMode

File: modules/hubspot/src/module.ts [hunk 10]
```diff
@@ -159,9 +171,9 @@ async function searchContact(input: IFlowInput, args: { secret: CognigySecret, q
 				if (properties.indexOf("vid") === -1) properties.push("vid");
 				if (contacts) {
 					contacts.forEach((res) => {
-						Object.keys(res).forEach((key) => {
+						Object.keys(res["properties"]).forEach((key) => {
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `res.properties`

File: modules/hubspot/src/module.ts [hunk 11]
```diff
@@ -159,9 +171,9 @@ async function searchContact(input: IFlowInput, args: { secret: CognigySecret, q
 				if (properties.indexOf("vid") === -1) properties.push("vid");
 				if (contacts) {
 					contacts.forEach((res) => {
-						Object.keys(res).forEach((key) => {
+						Object.keys(res["properties"]).forEach((key) => {
 							// if key isn't in the defined properties, delete it
-							if (properties.indexOf(key) === -1) delete res[key];
+							if (properties.indexOf(key) === -1) delete res["properties"][key];
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `res.properties`

File: modules/hubspot/src/module.ts [hunk 12]
```diff
@@ -201,7 +213,9 @@ async function searchContact(input: IFlowInput, args: { secret: CognigySecret, q
 async function findCompanyByDomain(input: IFlowInput, args: { secret: CognigySecret, domain: string, writeToContext: boolean, properties: string, store: string, stopOnError: boolean }): Promise<any> {
 	if (!args.secret || !args.secret.apiKey) return Promise.reject("Secret not defined or invalid.");
 	if (!args.domain) return Promise.reject("No domain defined.");
+
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `hubspot.qs.propertyMode`

File: modules/hubspot/src/module.ts [hunk 13]
```diff
@@ -216,9 +230,9 @@ async function findCompanyByDomain(input: IFlowInput, args: { secret: CognigySec
 
 				if (res && Array.isArray(res) && res.length > 0) {
 					res.forEach((company) => {
-						Object.keys(company).forEach((key) => {
+						Object.keys(company["properties"]).forEach((key) => {
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `company.properties`

File: modules/hubspot/src/module.ts [hunk 14]
```diff
@@ -216,9 +230,9 @@ async function findCompanyByDomain(input: IFlowInput, args: { secret: CognigySec
 
 				if (res && Array.isArray(res) && res.length > 0) {
 					res.forEach((company) => {
-						Object.keys(company).forEach((key) => {
+						Object.keys(company["properties"]).forEach((key) => {
 							// if key isn't in the defined properties, delete it
-							if (properties.indexOf(key) === -1) delete company[key];
+							if (properties.indexOf(key) === -1) delete company["properties"][key];
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `company.properties`

File: modules/hubspot/src/module.ts [hunk 15]
```diff
@@ -258,7 +272,9 @@ async function findCompanyByDomain(input: IFlowInput, args: { secret: CognigySec
 async function createCompany(input: IFlowInput, args: { secret: CognigySecret, data: any, writeToContext: boolean, properties: string, store: string, stopOnError: boolean }): Promise<any> {
 	if (!args.secret || !args.secret.apiKey) return Promise.reject("Secret not defined or invalid.");
 	if (!args.data) return Promise.reject("No data defined.");
+
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `hubspot.qs.propertyMode`

File: modules/hubspot/src/module.ts [hunk 16]
```diff
@@ -290,16 +306,17 @@ async function createCompany(input: IFlowInput, args: { secret: CognigySecret, d
...
  * @arg {Boolean} `stopOnError` Whether to stop on error or continue
  */
-async function updateCompany(input: IFlowInput, args: { secret: CognigySecret, companyId: number, data: any, writeToContext: boolean, properties: string, store: string, stopOnError: boolean }): Promise<any> {
+async function updateCompany(input: IFlowInput, args: { secret: CognigySecret, companyId: number, data: any, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<any> {
 	if (!args.secret || !args.secret.apiKey) return Promise.reject("Secret not defined or invalid.");
 	if (!args.companyId) return Promise.reject("No companyId defined.");
 	if (!args.data) return Promise.reject("No data defined.");
+
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `hubspot.qs.propertyMode`

File: modules/hubspot/src/module.ts [hunk 17]
```diff
@@ -337,7 +354,9 @@ async function updateCompany(input: IFlowInput, args: { secret: CognigySecret, c
 async function createEngagement(input: IFlowInput, args: { secret: CognigySecret, data: any, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<any> {
 	if (!args.secret || !args.secret.apiKey) return Promise.reject("Secret not defined or invalid.");
 	if (!args.data) return Promise.reject("No data defined.");
+	
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `hubspot.qs.propertyMode`

File: modules/hubspot/src/module.ts [hunk 18]
```diff
@@ -373,7 +392,9 @@ async function createEngagement(input: IFlowInput, args: { secret: CognigySecret
  */
 async function getOwners(input: IFlowInput, args: { secret: CognigySecret, writeToContext: boolean, store: string, stopOnError: boolean }): Promise<any> {
 	if (!args.secret || !args.secret.apiKey) return Promise.reject("Secret not defined or invalid.");
+
 	const hubspot = new Hubspot({ apiKey: args.secret.apiKey });
+	if (hubspot.qs && typeof hubspot.qs === 'object') hubspot.qs["propertyMode"] = 'value_only';
```
Comment (robinschuster) [inline modules/hubspot/src/module.ts]: `hubspot.qs.propertyMode`

File: modules/hubspot/README.md (hunk 1 above)
Comment (AlexTeusz) [reply modules/hubspot/README.md]: done

File: modules/hubspot/src/module.ts (hunk 2 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 3 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 4 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: I think there was an error, that's why i wrote it like this

Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 14 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 16 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 18 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 17 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 16 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 15 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 14 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

File: modules/hubspot/src/module.ts (hunk 5 above)
Comment (AlexTeusz) [reply modules/hubspot/src/module.ts]: done

**Comment by AlexTeusz** [comment]
I added all requested changes by @robinschuster

**Pull request merged**